src/main/java/de/tum/in/www1/artemis/service/programming/TemplateUpgradePolicyService.java

```diff
@@ -32,7 +32,7 @@ public TemplateUpgradePolicyService(JavaTemplateUpgradeService javaRepositoryUpg
     public TemplateUpgradeService getUpgradeService(ProgrammingLanguage programmingLanguage) {
         return switch (programmingLanguage) {
             case JAVA -> javaRepositoryUpgradeService;
-            case KOTLIN, PYTHON, C, HASKELL, VHDL, ASSEMBLER, SWIFT, OCAML, EMPTY -> defaultRepositoryUpgradeService;
+            default -> defaultRepositoryUpgradeService;
```

Comment:
we want to avoid default in switch expressions, because the compiler would not warn us if new enum values are added

Reply:
Corrected in [103f8fa](https://github.com/ls1intum/Artemis/pull/9074/commits/103f8fa2b46e4efb8a960aae6db1dfdc1e8809e6).